server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -291,49 +292,48 @@ export class AssetRepository implements IAssetRepository {
    * @returns Promise<string[]> - Array of assetIds belong to the device
    */
   async getAllByDeviceId(ownerId: string, deviceId: string): Promise<string[]> {
-    const rows = await this.assetRepository.find({
+    const items = await this.assetRepository.find({
+      select: { deviceAssetId: true },
       where: {
         ownerId,
         deviceId,
         isVisible: true,
+        deviceAssetId: Not(IsNull()),
```

Comment:
The `deviceAssetId` check seems unnecessary, because it's not nullable

Reply:
Thanks, removing it